### PR TITLE
Draft user manual.

### PR DESCRIPTION
**Pull request by jmlsf**
This is a preliminary draft of a "React User Manual", which does the following:

1. Gathers all of the information strewn about the release notes
2. Incorporates quite a bit of the information in the /docs directory (I hope that's okay)
3. Adds a section on React interop
4. Attempts to deal with #249 #166 and to a lesser extent #209 and #302

There are quite a few "TODO"s in this documentation where I just don't know the answer and it probably needs another pass through for spelling, but this should get the ball rolling.  Comments welcome.

A few comments on the technology discussion of #209: it seems like the two goals of a different documentation tool are (1) sidebar contents like in gistbook, asciidoctor, or slate and (2) live examples.  My only real comment is that it seems undeniable that the previous format was not really working (even the recent release notes don't use the custom format of the 0.6 and earlier release notes), and I would vote for some kind of markdown-based format.  Perhaps a modern tool + a post processor is the way to go, as people can just edit the markdown and forget about the live examples.  At any rate, I'd rather just get the content in a usable state, so if you can fill in some of the holes, please do.

### CODE REVIEW

File: docs/ReactUserGuide.md [hunk 1]
```diff
@@ -0,0 +1,1052 @@
...
+
+```clojure
+[:div>p>b "Nested Element"]
+```
+
+## Rendering Hiccup
+
+The primary entrypoint to the reagent library is `reagent.core/render`.
+
+```
```
Comment (gadfly361) [inline docs/ReactUserGuide.md]: add `clojure` after the backticks

File: docs/ReactUserGuide.md [hunk 2]
```diff
@@ -0,0 +1,1052 @@
...
+
+So the rookie mistake is to forget to put in the `[a b c]` parameters on the inner render function.
+
+Remember, `outer` is called once _per component instance_. Each time, the parameters to `outer` will hold the initial parameter values. The renderer on the other hand, will be called by Reagent many times and, each time, potentially with alternative parameter values, but unless you repeat the parameters on the renderer it will close over those initial values in `outer`. As a result, the component renderer will stubbornly only ever render the original parameter values, not the updated ones, which can be baffling for a beginner.
+
+## How Reagent Knows the Difference Between Form-1 and Form-2
+
+When Reagent is turning your Hiccup into React components, (see below for a fuller discussion of the rendering process), it will first invoke the form-1 or form-2 function specified. For example:
+
+```
```
Comment (gadfly361) [inline docs/ReactUserGuide.md]: add clojure after the backticks

**Comment by mike-thompson-day8** [comment]
@jmlsf  I greatly appreciate the effort. I know how long docs take. 

Within `/docs` we have a README which acts as something of Table of Contents, and then one doc for each "chapter".    I'd would prefer to continue with that process because it gives us incremental improvements.  (Rather than the big bang approach, of which I'm suspicious). 

I would suggest that we don't need to rewrite or replace the existing docs, but rather supplement them.  I'm definitely keen to focus on the words and worry about fancy rollup later.  

For example, as a first step, I'd be keen to see your "Using Hiccup to Describe HTML" section taken out into a separate document and put into a document called `UsingHiccupToDescribeHTML.md` and added to the README ahead of the existing docs.  It contains good information. 

Then, as a next step, let's identify the further documents we should add one at a time.  Let's do a PR for each one, and get them in incrementally. 

When we have a directory full of all the right material, one document representing each subject, then let's think about one rolled up document.  The incremental approach works best.  Let's not rewrite anything we already have.

**Comment by jmlsf** [comment]
I'm fine with this.  One comment about re-writing: I would suggest that we need to distinguish between tutorial / didactic type documents and reference material.  Reagent, although it describes itself as a "simple" interface to React, is actually really complicated and subtle, and one of its biggest flaws is that there is no reference material.  So to the extent I edited down some of the existing material, it was to that end.

**Comment by mike-thompson-day8** [comment]
Both kinds of docs (tutorial and reference) are useful.  I've tended to be focused on the tutorial kind. I've assumed that your PR was oriented that way too.

**Comment by mike-thompson-day8** [comment]
Many Thanks!

**Comment by pesterhazy** [comment]
Just saw this, great work @jmlsf!

**Pull request merged**